src/python/grpcio_channelz/grpc_channelz/v1/channelz.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2018 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Channelz debug service implementation in gRPC Python."""
+
+import grpc
+from grpc._cython import cygrpc
+
+import grpc_channelz.v1.channelz_pb2 as _channelz_pb2
+import grpc_channelz.v1.channelz_pb2_grpc as _channelz_pb2_grpc
+
+from google.protobuf import json_format
+
+
+class ChannelzServicer(_channelz_pb2_grpc.ChannelzServicer):
+    """Servicer handling RPCs for service statuses."""
+
+    # pylint: disable=no-self-use
+    def GetTopChannels(self, request, context):
+        try:
+            return json_format.Parse(
```

Comment:
What happens if this parse fails? Is an exception raised?

This would only happen if core was returning misformattesd JSON, but I would rather be safe agains that it possible. C++ layer handles bad JSON and returns INTERNAL

Reply:
I believe it will be caught [here](https://github.com/grpc/grpc/blob/aa00e06b65c2e238e6c178134f964ea269a2231b/src/python/grpcio/grpc/_server.py#L398) in the python server implementation and, by default, converted to UNKNOWN. Matching C++'s behavior here and returning INTERNAL here sounds good. json_format.Parse will throw a [`ParseError`](https://github.com/protocolbuffers/protobuf/blob/f22be4ddb0c8b8f81f84d106c74a812ee87f4c6b/python/google/protobuf/json_format.py#L395) which can be caught here.

Reply:
Adding another catch clause for `ParseError` which returns INTERNAL.